### PR TITLE
FPS Fix without Major Conviva API Changes

### PR DESCRIPTION
**Pull request by realeyes-mike-patterson**
Fixes
* Report the fps to the Conviva SDK without the API change

### CODE REVIEW

**Review by stonko1994 — APPROVED** [review]
Just to double-check this PR is a precondition for #29?

_Approving already_

**Comment by realeyes-mike-patterson** [comment]
> Just to double-check this PR is a precondition for #29?
> 
> _Approving already_

Nope. Sorry that was a poorly worded title. I'll adjust it.

This PR makes the "Average FPS" fix with a minor adjustment to existing code if it needs to ship sooner. PR #29 includes the API update for Conviva 4 so it's a major change.

**Comment by stonko1994** [comment]
> > Just to double-check this PR is a precondition for #29?
> > _Approving already_
> 
> Nope. Sorry that was a poorly worded title. I'll adjust it.
> 
> This PR makes the "Average FPS" fix with a minor adjustment to existing code if it needs to ship sooner. PR #29 includes the API update for Conviva 4 so it's a major change.

Gotcha 👍 Thx for clarification 🙂

File: conviva/src/main/java/com/bitmovin/analytics/conviva/ClientMeasureInterface.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+    public void setFrameRate(float frameRate) {
+        this.frameRate = frameRate;
+    }
+
+    // IClientMeasureInterface
+
+    @Override
+    public int getFrameRate() {
+        return Math.round(frameRate);
```
[the remark this answers is not in the document]
Comment (realeyes-mike-patterson) [reply conviva/src/main/java/com/bitmovin/analytics/conviva/ClientMeasureInterface.java:17]: I can use that. I've been using getVideoPlaybackData( ).getFrameRate( )
Should I make that change?

[the remark this answers is not in the document]
Comment (realeyes-mike-patterson) [reply conviva/src/main/java/com/bitmovin/analytics/conviva/ClientMeasureInterface.java:17]: With this PR I wanted to work within your existing code to limit potential side effects. So yes, you're correct, framerate is only updated when the other video quality metrics are updated in updateSession( )

[the remark this answers is not in the document]
Comment (luckygoyal-bitmovin) [reply conviva/src/main/java/com/bitmovin/analytics/conviva/ClientMeasureInterface.java:17]: ok, thanks for clarifying.

**Review by luckygoyal-bitmovin — APPROVED** [review]

**Comment by luckygoyal-bitmovin** [comment]
Hi @realeyes-mike-patterson I have taken the liberty to merge the fix so that I can create a release and provide to customer:)

**Pull request merged**